src/robusta/core/sinks/robusta/dal/supabase_dal.py

```diff
@@ -96,27 +104,79 @@ def __init__(
         self.sign_in()
         self.sink_name = sink_name
         self.signing_key = signing_key
+        self.event_loop = event_loop
+
+    def __to_db_scanResult(self, scanResult: ScanReportRow) -> Dict[Any, Any]:
+        db_sr = scanResult.dict()
+        db_sr["account_id"] = self.account_id
+        db_sr["cluster_id"] = self.cluster
+        return db_sr
+    
+
+    def persist_scan(self, enrichment: Enrichment):
+
+        for block in enrichment.blocks:
+            if not isinstance(block, ScanReportBlock):
+                return
+
+        db_scanResults = [self.__to_db_scanResult(sr) for sr in block.results]
+        res = self.client.table(SCANS_RESULT_TABLE).insert(db_scanResults).execute()
+        if res.get("status_code") not in [200, 201]:
+            msg = f"Failed to persist scan {block.scan_id} error: {res.get('data')}"
+            logging.error(msg)
+            self.handle_supabase_error()
+            raise Exception(msg)
+
+        self.client.postgrest.session.headers.update(self.client._get_auth_headers())
+        tasks = [self.event_loop.create_task(self.client.rpc("insert_scan_meta", {
```

Comment:
we added recently a sync rpc call impl
https://github.com/robusta-dev/robusta/blob/1d366f882cbe064ef70da4f5f2bad9ead7351cd6/src/robusta/core/sinks/robusta/dal/supabase_dal.py#L335

I think it's simpler and safer to use